spec.html

```diff
@@ -6460,6 +6460,23 @@ <h1>
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-sortstringlistbycodeunit" type="abstract operation">
+      <h1>
+        SortStringListByCodeUnit (
+          _strings_: a List of Strings,
+        ): a List of Strings
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>The returned List contains the same Strings as _strings_, but sorted lexicographically by UTF-16 code unit in ascending order.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _result_ be a copy of _strings_.
+        1. [declared="comparefn"] Sort _result_ into the same order as if an Array of the same values had been sorted using %Array.prototype.sort% with *undefined* as _comparefn_.
```

Comment:
This is great, and I'd love to see it taken even further by building the methods on top of AOs rather than the other way around:
* Introduce new AO `SortList(_list_, _SortCompare_)`, replacing [SortIndexedProperties](https://tc39.es/ecma262/multipage/indexed-collections.html#sec-sortindexedproperties) step 4 with a call to it.
* Introduce new AO `CompareStringsByCodeUnit(_x_, _y_)`, replacing [CompareArrayElements](https://tc39.es/ecma262/multipage/indexed-collections.html#sec-comparearrayelements) step 7 onwards with a call to it (and possibly also updating IsLessThan to use it).
* Redefine this SortStringListByCodeUnit AO to something like
  ```
  1. Let _SortCompare_ be a new Abstract Closure with parameters (_x_, _y_) that performs the following steps when called:
    1. Return CompareStringsByCodeUnit(_x_, _y_).
  1. Return SortList(_strings_, _SortCompare_)
  ```

Reply:
That's deeper than I had hoped to dig with this PR; how strong is that preference? If there's broad support for that direction I'm happy to do it. But if it's likely to result in back and forth, I'd prefer to keep the smaller scope and open an issue to discuss further changes.

Reply:
I have no objection to pursuing that in a followup if it is contentious.